### DISCUSSION TITLE
Osu! For VR

### DISCUSSION THREAD

**Discussion by levavft** [ppy/osu#15537] [Ideas] [answered]
Hi everyone, I hope this is the right place for this.

I was wondering if there has been any discussion about getting Osu! to VR, as a replacement of beatsaber.
It's probably a huge undertaking of course, so currently I'm only talking about the will to eventually do something like this.

Why is this even needed?
Well, mainly because Beatsaber doesn't really allow the community to create songs.
There is of course BMFB, but it exists despite Beat Games (the company).

Actually, if you have intentions to do something like this, you could maybe get help from the BMFB community, I would think that thei'd love have a platform that actually wants user created songs :)

that being said, I honestly don't know the programming communities of BMFB or of Osu!. I'm just a player of both games that would love to have the best of both worlds ;)

**Comment by nahkd123** [comment, marked as answer]
Someone actually forked osu! and created the VR version, however I can't seem to find it.

KoiFishu have made a video reviewing osu!lazer in VR, which you can find it here: https://www.youtube.com/watch?v=RiXeOb7-q8A

**Reply by levavft** [reply]
thank you! found a discord server with the binaries, it'll be open source again in a few months as well :)
would you consider merging something like this?

**Reply by smoogipoo** [reply]
> would you consider merging something like this?

Yes, we would.

**Reply by Flutterish** [reply]
Hey, just as a technicality, osu!xr is not a fork of osu itself, but it builds on top of osu!framework (as 2 separate projects - osu!xr and osu!framework-xr) and only uses osu as an embedded game, with a tiny bit of reflections to integrate some features such as ruleset action bindings.
